src/ol/format/EsriJSON.js

```diff
@@ -256,10 +256,10 @@ function readGeometry(object, opt_options) {
     const rings = convertRings(esriJSONPolygon.rings, layout);
     if (rings.length === 1) {
       type = GeometryType.POLYGON;
-      object['rings'] = rings[0];
+      object = Object.assign({}, object, {rings: rings[0]});
```

Comment:
The `rings` property was quoted here to stop minifiers from renaming it.  It could be worth validating that this is still necessary, but without doing that validation, I would suggest leaving it quoted.

Reply:
That's good, I put quotes.